pkg/v14/kubeconfig/desired.go

```diff
@@ -9,24 +9,28 @@ import (
 	"github.com/giantswarm/operatorkit/client/k8srestconfig"
 	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/giantswarm/cluster-operator/pkg/label"
 	"github.com/giantswarm/cluster-operator/pkg/v14/key"
-	awskey "github.com/giantswarm/cluster-operator/service/controller/aws/v14/key"
 )
 
 func (r *StateGetter) GetDesiredState(ctx context.Context, obj interface{}) ([]*corev1.Secret, error) {
-	cr, err := awskey.ToCustomObject(obj)
+	clusterGuestConfig, err := r.transformFunc(obj)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	if awskey.IsDeleted(cr) {
+	metaObject, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
```

Comment:
I don't think the use of the `meta package` here is a good idea.

In other resources we already have `ToClusterObjectMetaFunc`. The code will be more consistent if we use the same approach here.

e.g.  `https://github.com/giantswarm/cluster-operator/blob/542548deb3e32b8d3257b529293f95f7f59c7d37/pkg/v14/resource/namespace/resource.go#L43`

Reply:
This was my suggestion. Reason is that I do not see why we need the function to be configured when `meta.Accessor` gives us what we need. I would have done something like this though. 

```
func isDeleted(obj interface{}) (bool, error)
```

Just a private function within the package here and it takes some accessor. The meta magic can be done within and the deletion timestamp check can be done there as well.

Reply:
OK thanks for the explanation.

Since `kubeconfig` is the first new resource for supporting `app` CRs I'm fine with improving the design.

I like having a`isDeleted` function for this.